src/game_api/savestate.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+struct StateMemory;
+
+class SaveState
+{
+  public:
+    /// Create a new temporary SaveState/clone of the main level state. Unlike save_state slots that are preallocated by the game anyway, these will use 32MiB a pop and aren't freed automatically, so make sure to clear them or reuse the same one to save memory.
```

Comment:
This sounds kind of ominously, like if you don't clear it, you get memory leak

Reply:
Well I guess the backend could do that for you like for the predefined slots, but we have to be very sure that these are worthless after the level has been unloaded...

Reply:
I was thinking more like, if you just set the variable to nil or overwrite it, it will eventually just clear itself, so the clear is not really useful for people who will make like 2/3 slots, only for stuff like TAS that would make a ton of them

Reply:
Also question: how much did you test the loading save state from previous level? can you just warp to that level and load the saved state or is there something odd that it crashes on?

Reply:
Well the garbage collector does clear them too, it's just not in any hurry to do that automatically...

Reply:
> Also question: how much did you test the loading save state from previous level? can you just warp to that level and load the saved state or is there something odd that it crashes on?

I have tried replaying the same seed and loading an earlier state in an identical level, still crashed. I don't actually understand why or why not that should work, but that's one reason this option is still left there, for someone to find a way to do it.

Reply:
I think I tried both modes, but at least real seeded.